mediator/src/main/kotlin/pl/edu/prz/kod/mediator/adapters/http/oauth/Jwt.kt

```diff
@@ -32,13 +37,14 @@ class Auth0Jwt(private val secret: String) : Jwt {
         }
     }
 
-    override fun verify(token: AccessToken): Boolean {
+    override fun verify(token: AccessToken): TokenVerificationResult {
         try {
             val decoded = verifier.verify(token.value)
-            logEvent(TokenVerifiedEvent(decoded.claims["email"]?.asString() ?: "NONE"))
-            return true
+            val email = decoded.claims["email"]?.asString() ?: "NONE"
+            logEvent(TokenVerifiedEvent(email))
+            return TokenVerificationResult.Success(email)
```

Comment:
Shouldn't the `TokenVerificationResult.Failure` be returned here if `email == "NONE"` (so for some reason email couldn't be extracted from token)?

Reply:
You're right